### DISCUSSION TITLE
VS 2019 fails on Formatting of non-void pointers is disallowed (C2338)

### DISCUSSION THREAD

**Discussion by zkheraj** [gabime/spdlog#2745] [General]
Trying to evaluate between Spdlog and G3log and our in-house logging utilities.
Was able to compile but getting Formatting of non-void pointers is disallowed.

Are we able to print char* or wchar_t* pointers ?

**Comment by tt4g** [comment]
Duplicate #1492

Should use `fmt::ptr()`.

See fmt API document: https://fmt.dev/9.1.0/api.html#_CPPv4I0EN3fmt3ptrEPKv1T

**Reply by zkheraj** [reply]
Thank you for the tip.

**Reply by zkheraj** [reply]
With fmt::ptr the address is printed. Can it print the value at that address ?

     wchar_t wsz[MAX_PATH] = L"This is a wide-character string";
     char sz[MAX_PATH] = "This is a character string";

     Test_wchar(wsz);
     Test_char(sz);

void Test_wchar(wchar_t* pwsz)
{
   DF_TRACE_LOG_CONSOLE(L"Test_wchar printing {}", fmt::ptr(pwsz));
}

void Test_char(char* psz)
{
   DF_TRACE_LOG_CONSOLE("Test_char printing {:p}", fmt::ptr(psz));
}

OUTPUT:
[2023-06-02 14:36:21] [17988][28144][T] :: Test_wchar printing 0x136f730
[2023-06-02 14:36:21] [17988][28144][T] :: Test_char printing 0x136f620

**Reply by tt4g** [reply]
The `char *` should be able to be output.

```cpp
void log(char* str)
{
    spdlog::info("{}". str);
}
```

> **Note**
> If you want to add `char*` directly to the first argument, you must use `fmt::runtime()`.
> See #2473

`wchar_t *` must define an `SPDLOG_WCHAR_TO_UTF8_SUPPORT` macro.
However, this feature supports only Windows OS.

https://github.com/gabime/spdlog/blob/31cf79a70dfa61a4b2b7509eecca6a3b269bfc4a/include/spdlog/tweakme.h#L90-L94
